src/routes/votings/[id]/+page.svelte

```diff
@@ -1,9 +1,495 @@
+	import BillCategoryTag from '$components/BillCategoryTag/BillCategoryTag.svelte';
+	import BillCard from '$components/BillCard/BillCard.svelte';
+	import { DefaultVoteOption, type CustomVoteOption, DefaultVotingResult } from '$models/voting.js';
+	import { onMount } from 'svelte';
 	export let data;
-</script>
 
-<a href="{data.voting.id}/votes">Votes page</a>
+	let open = false;
+	let selectedTab = '';
+	let selectedMenu = 'summary';
+	let isViewPercent = false;
+	let searchQuery = '';
+	let isMobile = false;
+
+	$: totalVote = data.results.reduce((acc, vote) => acc + vote.total, 0);
+
+	interface AnchorElement extends HTMLElement {
+		offsetTop: number;
+		offsetHeight: number;
+	}
+
+	function dateConvertor(date: Date) {
+		const convertedDate = Intl.DateTimeFormat('th', {
+			dateStyle: 'medium',
+			calendar: 'buddhist'
+		}).format(date);
+		return convertedDate;
+	}
+
+	function getVoteColor(vote: DefaultVoteOption | CustomVoteOption | string) {
+		switch (vote) {
+			case DefaultVoteOption.Agreed:
+				return 'bg-teal-50';
+			case DefaultVoteOption.Disagreed:
+				return 'bg-red-50 text-white';
+			case DefaultVoteOption.Novote:
+				return 'bg-gray-80 text-white';
+			case DefaultVoteOption.Abstain:
+				return 'bg-gray-50';
+			case DefaultVoteOption.Absent:
+				return 'bg-gray-20';
+			default:
+				if (typeof vote === 'string') {
+					return 'bg-purple-70';
+				} else {
+					return `bg-purple-70 opacity-${vote.colorIntensity * 100} text-white`;
+				}
+		}
+	}
+
+	function getBillStatusColor(status: DefaultVotingResult | string) {
+		switch (status) {
+			case DefaultVotingResult.Passed:
+				return 'bg-teal-10';
+			case DefaultVotingResult.Failed:
+				return 'bg-red-10';
+			default:
+				return 'bg-purple-10';
+		}
+	}
+
+	function getWidthPercent(voteCount: number, totalVote = 750) {
+		return Math.round((voteCount / totalVote) * 100);
+	}
+
+	function scrollTo(id: string) {
+		selectedMenu = id;
+		const el = document.getElementById(id);
+		if (el) el.scrollIntoView({ behavior: 'smooth' });
+	}
+
+	function onScroll() {
+		const scrollY = window.scrollY;
+		const anchorEls = document.querySelectorAll<AnchorElement>('#summary, #byParty, #byPerson');
+
+		for (const el of anchorEls) {
+			const top = el.offsetTop;
+			const bottom = top + el.offsetHeight;
 
-<div class="whitespace-pre">
-	{JSON.stringify(data, undefined, 2)}
+			if (scrollY >= top && scrollY <= bottom) {
+				selectedMenu = el.id.replace('#', '');
+				break;
+			}
+		}
+	}
+
+	function getDiffDays(targetDate: Date): number {
+		const currentDate = new Date();
+
+		const timeDifference = currentDate.getTime() - targetDate.getTime();
+
+		const daysDifference = timeDifference / (1000 * 60 * 60 * 24);
+
+		const roundedDaysDifference = Math.round(daysDifference);
+
+		return roundedDaysDifference;
+	}
+
+	onMount(() => {
+		window.addEventListener('scroll', onScroll, { passive: true });
+		isMobile = !window.matchMedia(`(min-width: 672px)`).matches;
```

Comment:
Responsive using JS is an expensive resource, and difficult to handle when the screen size is changed. What is the reason that CSS is not enough in this case?